### PR TITLE
Added ReadMe file for the OOP Folder.

### PR DESCRIPTION
**Pull request by sabiha22**
Added readme file that explains the basic concepts of object oriented programming with Java Script.

### CODE REVIEW

File: 5 Object Oriented Programming/OOP_readme.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+};<br>
+let hungryRabbit = {<br>
+    &nbsp;type: "hungry",<br>
+    &nbsp;speak<br>
+};<br>
+whiteRabbit.speak("Oh my ears and whiskers, " + "how late it's getting!"); <br>     // The white rabbit says 'Oh my ears and whiskers, how late it's getting!' <br>
+hungryRabbit.speak("I could use a carrot right now.");     //The hungry rabbit says 'I could use a carrot right now.'   <br>
+
+You can think of this as an extra parameter that is passed in a different way. If you want to pass it explicitly, you can use a function’s call method, which takes the ‘this’ value as its first argument and treats further arguments as normal parameters.<br><br>
+speak.call(hungryRabbit, "Tasty!");<br>
```
Comment (MadaZZ) [inline 5 Object Oriented Programming/OOP_readme.md:34]: # Object-Oriented Programming
Object-oriented programming, a set of techniques that use objects (and related concepts) as the central idea of program organization.
## Key Concepts:

**Encapsulation**

The core idea in object-oriented programming is to divide programs into smaller pieces and make each piece responsible for managing its own state. This way, some knowledge about the way a piece of the program works can be kept local to that piece. Someone working on the rest of the program does not have to remember or even be aware of that knowledge. Whenever these local details change, only the code directly around it needs to be updated. Different pieces of such a program interact with each other through interfaces, limited sets of functions or bindings that provide useful functionality at a more abstract level, hiding their precise implementation. Such program pieces are modelled using objects. Their interface consists of a specific set of methods and properties. Properties that are part of the interface are called public. The others, which outside code should not and cannot touch, are called private. Even though the language doesn&#39;t have this distinction built in, JavaScript programmers are successfully using this idea. Typically, the available interface is described in documentation or comments. It is also common to put an underscore (\_) character at the start of property names to indicate that those properties are private. The idea of separating interface from implementation is called encapsulation.

**Methods**

Methods are nothing more than properties that hold function values. This is a simple method:
````js
let rabbit = {};<br>
	rabbit.speak = function(line) {
	console.log(`The person says '${line}'`);<br>
};

rabbit.speak("I'm good.");    //The rabbit says I'm good.
````
Usually a method needs to do something with the object it was called on when a function is called as a method— looked up as a property and immediately called, as in object.method()— the binding called this in its body automatically points at the object that it was called on.
```js
function speak(line) {
console.log(`The ${this.type} rabbit says '${line}'`);
}
let whiteRabbit = {
    type: "white",
    speak
};
let hungryRabbit = {
    type: "hungry",
    speak
};

whiteRabbit.speak("Oh my ears and whiskers, " + "how late it's getting!"); <br>     // The white rabbit says 'Oh my ears and whiskers, how late it's getting!' <br>
hungryRabbit.speak("I could use a carrot right now.");     //The hungry rabbit says 'I could use a carrot right now.'   <br>

You can think of this as an extra parameter that is passed in a different way. If you want to pass it explicitly, you can use a function’s call method, which takes the ‘this’ value as its first argument and treats further arguments as normal parameters.<br><br>
speak.call(hungryRabbit, "Tasty!");<br>
```

Comment (MadaZZ) [reply 5 Object Oriented Programming/OOP_readme.md:34]: **That is how you should be formatting your markdown.**

**Review by MadaZZ — CHANGES_REQUESTED** [review]
Please change the markdown as suggested in the code review.
Here are the links you can refer to sharpen your markdown skills: 
- [GitHub - Mastering Markdown](https://guides.github.com/features/mastering-markdown/)
- [Medium Blog](https://medium.com/better-programming/master-markdown-in-ten-minutes-ce9721c51eb)
- [How to add code snippet in Markdown](https://www.freecodecamp.org/news/how-to-format-code-in-markdown/)

**Comment by sabiha22** [comment]
I have updated according to the requirements.

File: 5 Object Oriented Programming/OOP_readme.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
+# Oect Oriented Programming
```
Comment (MadaZZ) [inline 5 Object Oriented Programming/OOP_readme.md:1]: Check the spelling here!

Comment (sabiha22) [reply 5 Object Oriented Programming/OOP_readme.md]: Sorry! I have fixed it now.

File: 5 Object Oriented Programming/OOP_readme.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+#### **The instanceof operator**
+
+It is occasionally useful to know whether an object was derived from a specific class. For this, JavaScript provides a binary operator called instanceof.
+```javascript
+console.log( new SymmetricMatrix(2) instanceof SymmetricMatrix); // true
+console.log(new SymmetricMatrix(2) instanceof Matrix); // true
+console.log(new Matrix(2, 2) instanceof SymmetricMatrix); // false
+console.log([1] instanceof Array); // true
+```
+The operator will see through inherited types, so a SymmetricMatrix is an instance of Matrix. The operator can also be applied to standard constructors like Array. Almost every object is an instance of Object.
```
Comment (MadaZZ) [inline 5 Object Oriented Programming/OOP_readme.md:141]: The format and explanation look good! 👍🏼

**Review by MadaZZ — APPROVED** [review]
Well done! ⭐️

File: 5 Object Oriented Programming/OOP_readme.md (hunk 3 above)
Comment (sabiha22) [reply 5 Object Oriented Programming/OOP_readme.md:141]: Thanks a lot!

**Pull request merged**